### ISSUE TITLE
SharpZipLib exception, invalid header

### ISSUE THREAD

**Issue by emperor06** [Atvaark/FtexTool#7]
I'm getting a consistant crash while trying to convert a few specific ftex files. Those textures render well in-game. It seems that SharpZipLib doesn't like the header. While it may be a bug in SharpZipLib, I think it's best to let you know.

Here's the error (ftex file attached)

> Unhandled Exception: ICSharpCode.SharpZipLib.SharpZipBaseException: Header checksum illegal
>    at ICSharpCode.SharpZipLib.Zip.Compression.Inflater.DecodeHeader()
>    at ICSharpCode.SharpZipLib.Zip.Compression.Inflater.Decode()
>    at ICSharpCode.SharpZipLib.Zip.Compression.Inflater.Inflate(Byte[] buffer, Int32 offset, Int32 count)
>    at ICSharpCode.SharpZipLib.Zip.Compression.Streams.InflaterInputStream.Read(Byte[] buffer, Int32 offset, Int32 count)
>    at System.IO.Stream.InternalCopyTo(Stream destination, Int32 bufferSize)
>    at System.IO.Stream.CopyTo(Stream destination)
>    at FtexTool.ZipUtility.Inflate(Byte[] buffer) in FtexTool\ZipUtility.cs:line 12
>    at FtexTool.Ftexs.FtexsFileChunk.SetData(Byte[] chunkData, Boolean compressed) in FtexTool\Ftexs\FtexsFileChunk.cs:line 41
>    at FtexTool.Ftexs.FtexsFileChunk.Read(Stream inputStream, Int32 baseOffset) in FtexTool\Ftexs\FtexsFileChunk.cs:line32
>    at FtexTool.Ftexs.FtexsFileChunk.ReadFtexsFileChunk(Stream inputStream, Int32 baseOffset) in FtexTool\Ftexs\FtexsFileChunk.cs:line 18
>    at FtexTool.Ftexs.FtexsFileMipMap.Read(Stream inputStream, Int16 chunkCount,Int32 baseOffset) in FtexTool\Ftexs\FtexsFileMipMap.cs:line 58
>    at FtexTool.Ftexs.FtexsFileMipMap.ReadFtexsFileMipMap(Stream inputStream, Int16 chunkCount, Int32 baseOffset) in FtexTool\Ftexs\FtexsFileMipMap.cs:line 50
>    at FtexTool.Ftexs.FtexsFile.Read(Stream inputStream, Int16 chunkCount, Int32baseOffset) in FtexTool\Ftexs\FtexsFile.cs:line 35
>    at FtexTool.Program.GetFtexFile(String filePath) in FtexTool\Program.cs:line 182
>    at FtexTool.Program.UnpackFtexFile(String filePath, String outputPath) in FtexTool\Program.cs:line 138
>    at FtexTool.Program.Main(String[] args) in FtexTool\Program.cs:line 52

[texture.zip](https://github.com/Atvaark/FtexTool/files/467921/texture.zip)

**Comment by Atvaark** [comment]
SharpZipLib is working correctly. FtexTool does some assumptions to work with a majority of the files and this one gets handled incorrectly. This .ftex file has an unknown flag which just 3% of all the ftex files have set. The error could be related to not handing that flag correctly.

I had to change the code a little bit to convert that texture to DDS and back to ftex again. But I can't commit it right now as I'd have to test if the change breaks unpacking the remaining 97% of the files.
Result: [c681a9ba4f51.jpg](https://cloud.githubusercontent.com/assets/5101884/19941454/79909452-a130-11e6-9313-4fd3278d30fc.jpg) **NSFW**

See this patch file for the change I made:
[fix.zip](https://github.com/Atvaark/FtexTool/files/567428/fix.zip)
